src/Microsoft.AspNetCore.TestHost/TestServer.cs

```diff
@@ -27,6 +27,14 @@ public TestServer(IWebHostBuilder builder)
             _hostInstance = host;
         }
 
+        public TestServer(IWebHostBuilder builder, Action<TestServer> configureServer)
+        {
+            configureServer(this);
```

Comment:
It's hazardous to expose a type instance externally from its own constructor. The type initialization hasn't completed yet so the state is indeterminate. e.g. Host is still null and code that operates on a TestServer expects Host to be non-null.

It may be cleaner to have a separate TestServerOptions class that can use used to flow the state you want.